### ISSUE TITLE
npm ci failed with missing node-pre-gyp

### ISSUE THREAD

**Issue by yiochen** [node-gfx/node-canvas-prebuilt#50]
Step to reproduce:
```
> mkdir test-canvas-prebuilt
> cd test-canvas-prebuilt
> npm init
> npm install canvas-prebuilt
> npm ci
```
throws error:
```
npm WARN prepare removing existing node_modules/ before installation

> canvas-prebuilt@1.6.5-prerelease.1 install some-path/node_modules/canvas-prebuilt
> node_modules/.bin/node-pre-gyp install

sh: node_modules/.bin/node-pre-gyp: No such file or directory
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! canvas-prebuilt@1.6.5-prerelease.1 install: `node_modules/.bin/node-pre-gyp install`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the canvas-prebuilt@1.6.5-prerelease.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     some-path/.npm/_logs/2018-08-17T23_32_55_765Z-debug.log
```

From [this](https://github.com/grpc/grpc-node/issues/365#issuecomment-391578068), it seems like we need to list `node-pre-gyp` as bundledDependencies in package.json

node -v
10.3.0
npm -v
6.1.0

**Comment by chearon** [comment]
Shouldn't having it in `dependencies` ensure `node-pre-gyp` is present? It sounds like `bundledDependencies` copies dependencies into the tarball before publishing, but we shouldn't have to force people to have multiple copies of `node-pre-gyp`.

**Comment by yiochen** [comment]
I am also confused. This seems to be an issue with `npm ci`.
I managed to work around it by adding `node-pre-gyp` to my package.json

**Comment by chearon** [comment]
Yeah... seems like a bug in `npm`. Feel free to re-open if you find otherwise.

**Comment by Pumpuli** [comment]
Turns out the install script doesn't work at all on Windows native command line due to it having forward slashed and no quotes. This was of course the original reason for #13, it just also happened to fix the missing `node-pre-gyp` issue.

@chearon could this fix be backported to `latest` package too (i.e. 1.x)?

**Comment by chearon** [comment]
Yeah sorry, I'll try to do that soon, someone else just asked for it in #51.

**Comment by chearon** [comment]
Sorry I missed that, it's been difficult supporting 1.x which is an old/broken build and the 2.x series and build.

I pushed the commit @Pumpuli mentioned to the 1.x build branch for the next release. As for a 1.6.12, version of `canvas-prebuilt`, I screwed that up a while back and unpublished it because it accidentally included `canvas` as a dependency 🤦‍♂️ so should I:

* Release a `canvas-prebuilt@1.6.13` that's really `canvas@1.6.12`
* Ask @LinusU to create a new `canvas@1.6.13` noop release so I can create a `canvas-prebuilt@1.6.13`?

I'm not sure which is worse
